classes/order/Order.php

```diff
@@ -1607,6 +1607,21 @@ public static function getByReferenceAndEmail($reference, $email)
 
         return new Order($id);
     }
+    
+    public static function getByReferenceEmailAndNumber($reference, $email, $ordernumber)
```

Comment:
I am not sure about the naming : ordernumber, following arguments naming model in this same file, I think `$order_number` is better

Reply:
Perhaps the same SQL of the previous method can be  reused somehow avoiding code duplicate.

Reply:
You have to type hint the method.